template/workflows/helm/.github/workflows/azure-kubernetes-service-helm.yml

```diff
@@ -131,5 +135,10 @@ jobs:
 
       - name: Deploy application on public cluster
         if: steps.isPrivate.outputs.PRIVATE_CLUSTER != 'true'
-        run: helm upgrade --wait -i -f ${{ env.CHART_OVERRIDE_PATH }} --set ${{ env.CHART_OVERRIDES }} --set image.tag=${{ github.sha }} automated-deployment ${{ env.CHART_PATH }} --namespace ${{ env.NAMESPACE }} --create-namespace
+        run: |
```

Comment:
breaking this into two separate steps with compound `if` clauses prevents the two layers of conditional control: gh action "if" and bash "if"

a "Deploy application on fleet cluster" step that only runs on fleet type with a "CLUSTER_RESOURCE_TYPE  == Microsoft.ContainerService/managedCluster guard on the old public step allows these to exist in parallel until the logic can be parameterized fully for the namespace creation